src/when-contexts.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from 'vscode';
+import { TokenCursor } from './cursor-doc/token-cursor';
+import * as docMirror from './doc-mirror'
+
+type CursorContext = 'calva-standard' | AltCursorContext;
+type AltCursorContext = 'string' | 'comment';
+const stringContext = 'calva:cursorInString';
+const commentContext = 'calva:cursorInComment';
+let lastContext: CursorContext = null;
+
+export default function setCursorContextIfChanged(editor: vscode.TextEditor) {
+    if (!editor || !editor.document || editor.document.languageId !== 'clojure') return;
+
+    const currentContext = determineCursorContext(editor.document, editor.selection.active);
+    setCursorContext(currentContext);
+}
+
+function setCursorContext(currentContext: CursorContext) {
+    if (lastContext === currentContext) {
+        return;
+    }
+    lastContext = currentContext;
+   
+    vscode.commands.executeCommand('setContext', commentContext, false);
+    vscode.commands.executeCommand('setContext', stringContext, false);
+
+    switch (currentContext) {
+        case 'calva-standard':
+            break;
+        case 'comment':
+            vscode.commands.executeCommand('setContext', commentContext, true);
+            break;
+        case 'string':
+            vscode.commands.executeCommand('setContext', stringContext, true);
+            break;
+        default:
+        // TODO: throw? log?
```

Comment:
I don't see that we would ever end up here. Which might indicate that we should log it. `console.warn()`, might be a good fit.

Reply:
Compile time is the real risk here (missing a case for a newly added context). Not being a Typescript maven, I was a bit puzzled here as I thought it would do an exhaustiveness check. I've found a better way to deal with it.